File: android/hakka-network-noop/src/main/kotlin/com/noodleapps/hakka/LogStore.kt
package com.noodleapps.hakka

import java.util.concurrent.locks.ReentrantReadWriteLock
import kotlin.concurrent.read
import kotlin.concurrent.write

/**
 * No-op LogStore: all methods are no-ops, returns empty/default values.
 * Thread-safe but does nothing.
 */
class LogStore(config: HakkaConfig = HakkaConfig()) {
    private val lock = ReentrantReadWriteLock()

    fun add(request: NetworkRequest) {
    }

    fun all(): List<NetworkRequest> = lock.read { emptyList() }

    fun get(id: String): NetworkRequest? = lock.read { null }

    fun size(): Int = lock.read { 0 }

    fun query(filter: RequestFilter): List<NetworkRequest> = lock.read { emptyList() }

    fun recent(count: Int): List<NetworkRequest> = lock.read { emptyList() }

    /** No-op — always returns false (nothing stored). */
    fun update(id: String, transform: (NetworkRequest) -> NetworkRequest): Boolean = false

    fun clear() = lock.write {
    }

    /** Always returns a zeroed summary — no requests captured in noop. */
    fun metricsSummary(): NetworkMetricsSummary = NetworkMetricsSummary(
        totalRequests = 0,
        completedRequests = 0,
        successCount = 0,
        errorCount = 0,
        averageResponseTimeMs = 0.0,
        successRate = 1.0,
        errorRate = 0.0,
        p95LatencyMs = null,
        totalDataTransferredBytes = 0L,
    )
}

File: ios/Sources/NetworkNoop/LogStore.swift
import Foundation
import HakkaCommon

/// No-op log store. All mutations are discarded, all queries return empty results.
public final class LogStore: @unchecked Sendable {
    /// Number of stored requests. Always 0.
    public var count: Int { 0 }

    /// All stored requests. Always empty.
    public var requests: [NetworkRequest] { [] }

    public init(capacity: Int = 500) {}

    /// No-op.
    public func add(_ request: NetworkRequest) {}

    /// Look up a request by ID. Always `nil`.
    public func request(byId id: String) -> NetworkRequest? { nil }

    /// No-op. Always returns `false` — the noop store holds no entries to mutate.
    @discardableResult
    public func update(id: String, transform: (inout NetworkRequest) -> Void) -> Bool { false }

    /// Query requests matching the given filter. Always empty.
    public func query(filter: RequestFilter) -> [NetworkRequest] { [] }

    /// No-op.
    public func clear() {}

    /// Returns the last `count` requests. Always empty.
    public func recent(_ count: Int) -> [NetworkRequest] { [] }

    /// No-op.
    public func removeOlderThan(_ epochMs: Int64) {}

    /// Returns an empty metrics summary. The noop store holds no requests.
    public func metricsSummary() -> NetworkMetricsSummary {
        NetworkMetricsSummary(
            totalRequests: 0, completedRequests: 0,
            successCount: 0, errorCount: 0,
            averageResponseTime: 0, successRate: 1.0, errorRate: 0,
            p95LatencyMs: nil, totalDataTransferred: 0
        )
    }

    /// Always `false` — the noop store holds no requests.
    public func containsRequestOlderThan(_ epochMs: Int64) -> Bool { false }
}
